### ISSUE TITLE
Update ideas

### ISSUE THREAD

**Issue by jonom** [kinglozzer/htmleditoriframe#2]
Hi mate, just tried this module out. Think it's a great idea but seems to need a few updates. Just wondering if you still use this module or if you're now using something else? Given the ubiquity of iframes for embedding content I think it could be worth polishing up this module and possibly merging it in to framework. What do you think? If you're keen I'm happy to help out.

Here are some ideas for updates:

* Forcing http:// protocol will break some iframes like Google Maps which require an SSL connection, so the protocol should be required as part of the src URL string instead of being appended automatically
* When services offer iframe embed code they offer a whole piece of code ready for copy and pasting. Would be great if this module could anticipate the pasting of such code and parse relevant properties from it.
* Enable the use of [intrinsic ratios](http://alistapart.com/d/creating-intrinsic-ratios-for-video/example2.html) for iframes that scale to fit available space and preserve aspect ratio (based on provided width and height). This should maybe be a checkbox as it would be great for a video but less desirable for an element with an adaptive layout such as a map.

**Comment by kinglozzer** [comment]
Yeah this really needs some love, it was something I originally threw together quickly for a client back in the 3.0 days and it hasn’t had much attention since! I’d definitely be happy to accept updates and improvements :)

> Forcing http:// protocol

That’d be because I copied and pasted the “Insert Media” code in my rush (which, strangely, still seems to force http when entering a URL - though oEmbed takes care of https for you)! It should probably just be a plain input instead with the “Add” button

> ...whole piece of code ready for copy and pasting

Great idea! We could offer two separate inputs - an “Insert Iframe URL” input and a “Paste Iframe code” textarea, it shouldn’t be too difficult to parse out attributes from the HTML

> intrinsic ratios

This one might be a little more tricky due to the extra CSS needed. I think we’d have to do it inline so that we can guarantee the exact ratio is preserved (rather than different classes for each possible ratio or something). Would we want to also support a max-width/height, assuming that’s possible? So it could be responsive/full-width up until the max width/height specified.

**Comment by jonom** [comment]
> We could offer two separate inputs - an “Insert Iframe URL” input and a “Paste Iframe code” textarea, it shouldn’t be too difficult to parse out attributes from the HTML

If possible I think it would be neat to just have the one iframe URL input, but have a description beside it that says you can paste complete iframe code there. Then when people paste complete code, it filters it down to just the URL, and updates any other relevant properties in the form such as width and height according to the pasted code.

> intrinsic ratios... This one might be a little more tricky due to the extra CSS needed

Yeah I think the thing to do here would just be to enable the possibility of this support, but would definitely need some consideration from the theme designer to make sure things work as intended. So as you suggest, we could inline the bare-minimum (which would be the padding-bottom amount which determines the ratio of the iframe) and include a class name like 'intrinsic' and let the rest be handled by stylesheets.

This kind of thing would benefit from shortcode placeholder support as [proposed here](https://github.com/silverstripe/silverstripe-framework/issues/4337). Then developers could modify the output of the iframe data themselves by changing a template.
